### PR TITLE
Add async sink for sending packets

### PR DESCRIPTION
**Pull request by jethrogb**
Add an implementation of `futures::Sink` to be able to send packets asynchronously. (UNIX only - I'm not too familiar with Windows).

The `Sink` API is a bit difficult to understand. This implementation is modeled after [`Unfold`](https://docs.rs/futures/0.3.31/futures/sink/struct.Unfold.html) and [`IntoSink`](https://docs.rs/futures/0.3.31/futures/io/struct.IntoSink.html).

### CODE REVIEW

File: src/sink/mod.rs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        cx: &mut task::Context<'_>,
+    ) -> Poll<Result<(), Self::Error>> {
+        let sink = Pin::into_inner(self);
+        let packet_storage = &mut sink.packet;
+
+        loop {
+            let mut guard = ready!(sink.inner.poll_write_ready_mut(cx))?;
+
+            if let Some(packet) = packet_storage.take() {
+                match guard.try_io(
```
Comment (raoulstrackx) [inline src/sink/mod.rs:86]: Nitpick: this `match` is a bit hard to parse due to the closure. Consider create a separate variable for it.

**Review by raoulstrackx — APPROVED** [review]

**Comment by Wojtek242** [comment]
@jethrogb Thanks for the PR!

Could you please do the following so that I can merge this PR:
1. Add a CHANGELOG entry,
2. Fix clippy lints
3. Add unit tests
